src/transformState.js

```diff
@@ -4,8 +4,30 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
-function transformState(state, actions) {
-  // write code here
+function modifyState(state, actions) {
+  for (const action of actions) {
+    const { type } = action;
+
+    if (type === 'addProperties') {
+      const { extraData } = action;
+
+      Object.assign(state, extraData);
+    } else if (type === 'removeProperties') {
+      const { keysToRemove } = action;
+
+      for (const key of keysToRemove) {
+        if (state.hasOwnProperty(key)) {
+          delete state[key];
+        }
+      }
+    } else if (type === 'clear') {
+      for (const key in state) {
+        if (state.hasOwnProperty(key)) {
+          delete state[key];
+        }
+      }
+    }
```

Comment:
Use switch statement if you have limited amount of conditions. switch (action.type) Look at the checklist.md